### ISSUE TITLE
Create new `Ads` module `Settings` PHP class

### ISSUE THREAD

**Issue by 10upsimon** [google/site-kit-wp#8223]
## Feature Description

The Ads module requires the addition of an applicable PHP Ads `Settings` class to the current codebase, sitting alongside the newly added base `Ads` class. This PHP settings class creation involves creation of a `Settings.php` class within `includes/Modules/Ads`. This should follow the same design pattern as all other module settings classes. 

See relevant section in design doc: https://docs.google.com/document/d/1APuSv95bf62uhzlaFlW6jPrzPKy1avpRYd9W1MSAAJo/edit?resourcekey=0-UuynlcUz9CoubgldR6Z5sg#heading=h.8bapw1nwueej 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Definition of a new `Settings.php` module settings class file at `includes/Modules/Ads/Settings.php`. See the design doc:  https://docs.google.com/document/d/1APuSv95bf62uhzlaFlW6jPrzPKy1avpRYd9W1MSAAJo/edit?resourcekey=0-UuynlcUz9CoubgldR6Z5sg#heading=h.8bapw1nwueej 
  * Class should be referenced as class `Settings` in the applicable `Ads` namespace
  * Class should extend the `Module_Settings` base class
  * Class should implement the `Setting_With_Owned_Keys_Interface` 
  * `OPTION` constant should be defined as the following value: 'googlesitekit_ads_settings'
  * Applicable owned keys within the `get_owned_keys` method should be defined: 'adsConversionID'
  * Values within the `get_default` method should be appropriately defined
* Update the `Ads` module class:
  * Class should use the `Module_With_Settings_Trait` trait
  * Class should implement the `Module_With_Settings` interface
  * Class should implement the `setup_settings` method

## Implementation Brief

* [x] Add `includes/Modules/Ads/Settings.php` class
  * [x] Use `includes/Modules/Search_Console/Settings.php` as a starting point
  * [x] Extend the `Module_Settings` base class
  * [x] It should implement the `Setting_With_Owned_Keys_Interface` 
  * [x] `OPTION` constant should be defined as the following value: `googlesitekit_ads_settings`
  * [x] Applicable owned keys within the `get_owned_keys` method should be defined: 'adsConversionID'
  * [x] It should introduce 1 setting for now in owned keys, and defaults - `adsConversionID` with default value `''`
* [x] Update `Google\Site_Kit\Modules\Ads` class:
  * [x] Use `Module_With_Settings_Trait`, and `Module_With_Settings` interface
  * [x] Implement `setup_settings`
  * [x] Add `is_connected` and `on_deactivation` methods, they can be borrowed from Analytics module, for example.
    * To check connection status, use `adsConversionID` setting
    * On deactivation delete the option name `googlesitekit_ads_settings` 

### Test Coverage

* Add `SettingsTest` class
* Update `Ads` class tests where applicable

## QA Brief

* No QA needed as there are no changes to QA/no user-facing changes.

## Changelog entry

* N/A

**Comment by eugene-manuilov** [comment]
@10upsimon the new settings class should be registered in the Ads module class that should be updated to implement the `Module_With_Settings` interface and to use the `Module_With_Settings_Trait` trait.

**Comment by 10upsimon** [comment]
That is correct @eugene-manuilov , I used the Search Console class references as examples. Would that level of technical detail not go into the IB? Or would it go into AC? I purposefully did not want to put too much "technical how to" in the AC portion, but let me know if I should?

**Comment by eugene-manuilov** [comment]
> I used the Search Console class references as examples. Would that level of technical detail not go into the IB? Or would it go into AC?

Ah... okay, I got it now. 

>  I purposefully did not want to put too much "technical how to" in the AC portion, but let me know if I should?

In my understanding, AC is a set of criteria that we need to use to determine whether a task is implemented correctly and completely or not. Usually, when i write it, I try to use statements that explain the final state that we want to achieve. Constructions like "Y should be done similarly to X" are quite vague and not really clear whether the final result satisfy this criteria or not.

For example, if we rephrase your criteria from:
```
The logic surrounding the settings class and its settings equivalent can largely be borrowed 
and implemented from a simpler module class, such as those defined within the Search_Console
module settings class at includes/Modules/Search_Console/Settings.php
```

To something like this:
```
The Ads module class registers settings by implementing the `Module_With_Settings` interface
and using the `Module_With_Settings_Trait` trait.
```

Then we get much more strict requirement and will be clear what the final result we want to achieve. WDYT?

**Comment by 10upsimon** [comment]
@eugene-manuilov does the updated AC read a little cleaner and more verbose to you, or do you believe there are still AC points missing?

**Comment by eugene-manuilov** [comment]
> * Class should use the `Module_With_Settings_Trait` trait

@10upsimon, this should be done in the Ads module class. Plus we need to add the `Module_With_Settings` interface to it and implement the `setup_settings` method in the module class.

I would suggest group requirements for different files. Something like this:

* Definition of a new `Settings.php` module settings class file at `includes/Modules/Ads/Settings.php`. See the design doc:  https://docs.google.com/document/d/1APuSv95bf62uhzlaFlW6jPrzPKy1avpRYd9W1MSAAJo/edit?resourcekey=0-UuynlcUz9CoubgldR6Z5sg#heading=h.8bapw1nwueej 
  * Class should be referenced as class `Settings` in the applicable `Ads` namespace
  * Class should extend the `Module_Settings` base class
  * Class should implement the `Setting_With_Owned_Keys_Interface` 
  * `OPTION` constant should be defined as the following value: 'googlesitekit_ads_settings'
  * Applicable owned keys within the `get_owned_keys` method should be defined: 'adsConversionID'
  * Values within the `get_default` method should be appropriately defined
* Update the Ads module class:
  * Class should use the `Module_With_Settings_Trait` trait
  * ...

**Comment by 10upsimon** [comment]
@eugene-manuilov back to you for AC approval again, thank you.

**Comment by eugene-manuilov** [comment]
Awesome! Thanks, @10upsimon. AC ✔️

**Comment by tofumatt** [comment]
@zutigrm Let's avoid using "Follow the guidelines from AC" in IB descriptions, just because in this case it's a bit vague.

For extremely simple/descriptive ACs it's fine, but as soon as it's more complex like this and there's other parts to the IB, I think it makes it harder to work out the IB plan. It'd be better to just copy and adjust the ACs here I think, which is fine to do in some cases.

What would the default value of `adsConversionID` be? I'm guessing a `null` value? 🤔

**Comment by zutigrm** [comment]
@tofumatt Thanks for the review. 

> Let's avoid using "Follow the guidelines from AC" in IB descriptions, just because in this case it's a bit vague.

I tend to avoid duplicating AC, as there is a section in our handbook dedicated for that point `Avoid duplicating the ACs in the IB`. AC has outlined the needed points so I referenced to it for that reason also. But in this case, I moved AC points to the appropriate parts in IB for sake of clarity.

> What would the default value of adsConversionID be? I'm guessing a null value?

We seems to use empty string for the most settings - some ID setting examples `clientID`, `accountID`, etc. So I updated IB to propose that as default value. 

Let me know what you think
